### PR TITLE
refactor pointerBitmap()

### PR DESCRIPTION
**Pull request by WalterBright**
Improved it a bit, not as much as I'd like to. It's still hard to read.

### CODE REVIEW

File: compiler/src/dmd/traits.d [hunk 1]
```diff
@@ -92,44 +93,50 @@ private Dsymbol getDsymbolWithoutExpCtx(RootObject oarg)
...
+        eSink.error(loc, "size overflow for type `%s`", t.toChars());
         return ulong.max;
     }
 
-    ulong bitsPerWord = sz_size_t * 8;
-    ulong cntptr = (sz + sz_size_t - 1) / sz_size_t;
-    ulong cntdata = (cntptr + bitsPerWord - 1) / bitsPerWord;
+    ulong bitsPerElement = sz_size_t * 8;  // bits used in each array element
+    ulong cntptr = (sz + sz_size_t - 1) / sz_size_t; // pointers have same size as sz_size_t
+    ulong length = (cntptr + bitsPerElement - 1) / bitsPerElement; // a bit per pointer
```
Comment (thewilsonator) [inline compiler/src/dmd/traits.d:126]: could these all be `const`?

File: compiler/src/dmd/traits.d [hunk 2]
```diff
@@ -282,28 +289,28 @@ ulong getTypePointerBitmap(Loc loc, Type t, Array!(ulong)* data)
...
     if (!t)
     {
-        error(e.loc, "`%s` is not a type", (*e.args)[0].toChars());
+        eSink.error(e.loc, "`%s` is not a type", (*e.args)[0].toChars());
         return ErrorExp.get();
     }
 
     Array!(ulong) data;
-    ulong sz = getTypePointerBitmap(e.loc, t, &data);
+    ulong sz = getTypePointerBitmap(e.loc, t, data, eSink);
```
Comment (thewilsonator) [inline compiler/src/dmd/traits.d:308]: ditto

**Review by thewilsonator — APPROVED** [review]
Looks good

File: compiler/src/dmd/traits.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/traits.d]: ok

File: compiler/src/dmd/traits.d (hunk 2 above)
Comment (WalterBright) [reply compiler/src/dmd/traits.d]: ok

**Pull request merged**